package/oracledbinstall.js

```diff
@@ -57,6 +58,36 @@ const PACKAGE_PATH_REMOTE = '/oracle/node-oracledb/releases/download/' + package
 const SHA_PATH_REMOTE = '/oracle/node-oracledb/releases/download/' + packageUtil.dynamicProps.GITHUB_TAG + '/' + packageUtil.SHA_FILE_NAME;
 const PORT = 443;
 
+function readProxyUrl(){
+  // Prefer env variable if exists
+  // This also serves as a fallback/workaround solution since
+  // getting npm config can be slow on some environments
+  const envProxy = process.env.https_proxy ||
+    process.env.HTTPS_PROXY ||
+    process.env.http_proxy ||
+    process.env.HTTP_PROXY ||
+    process.env.all_proxy ||
+    process.env.ALL_PROXY;
+
+  if(envProxy){
+    return envProxy;
+  }
+
+  const httpsProxy = execSync('npm config get https-proxy').toString().trim();
```

Comment:
Seems like we are favoring proxy settings from env over npm config. Is that standard? Should it be reversed?

Reply:
The ordering was chosen as a way to overcome the slow perf issues of `npm config get` without needing an 'emergency option', see https://github.com/oracle/node-oracledb/pull/919#issuecomment-397446225

Reply:
To explain further:
a) In order not to break current setups
b) Prevent blocking the installation due to any issues with `npm config get` (though I think it should work fine most of the time)